api/src/certification/configuration/infrastructure/repositories/flash-algorithm-configuration-repository.js

```diff
@@ -1,6 +1,38 @@
+/**
+ * @typedef {import ('../../../shared/domain/models/FlashAssessmentAlgorithmConfiguration.js').FlashAssessmentAlgorithmConfiguration} FlashAssessmentAlgorithmConfiguration
+ */
+
 import { DomainTransaction } from '../../../../shared/domain/DomainTransaction.js';
 
+/**
+ * @param {FlashAlgorithmConfiguration} flashAlgorithmConfiguration
+ * @returns {Promise<void>}
+ */
 export const save = async (flashAlgorithmConfiguration) => {
   const knexConn = DomainTransaction.getConnection();
-  return knexConn('flash-algorithm-configurations').insert(flashAlgorithmConfiguration.toDTO());
+
+  const currentActiveConfiguration = await knexConn('certification-configurations')
+    .select('id')
+    .whereNull('expirationDate')
+    .first();
+
+  const [{ startingDate: dateOfConfigurationSwitching }] = await knexConn('certification-configurations')
+    .insert({
+      startingDate: knexConn.fn.now(),
+      expirationDate: null,
+      challengesConfiguration: {
+        maximumAssessmentLength: flashAlgorithmConfiguration.maximumAssessmentLength,
+        challengesBetweenSameCompetence: flashAlgorithmConfiguration.challengesBetweenSameCompetence,
+        limitToOneQuestionPerTube: flashAlgorithmConfiguration.limitToOneQuestionPerTube,
+        enablePassageByAllCompetences: flashAlgorithmConfiguration.enablePassageByAllCompetences,
+        variationPercent: flashAlgorithmConfiguration.variationPercent,
+      },
+    })
+    .returning('startingDate');
+
+  if (currentActiveConfiguration?.id) {
```

Comment:
Ce `if` implique un peu de réflexion pour comprendre à quoi il sert, je propose de faire une fonction pour la nommer et rendre + agréable la lecture


```suggestion
  if (mustExpirePreviousConfiguration(existingActiveConfigurationBeforeInsert)) {
```

```js
const mustExpirePreviousConfiguration = (existingActiveConfigurationBeforeInsert) => {

return !!currentActiveConfiguration?.id;

}
```